lexer/impl/lexer.cpp

```diff
@@ -9,315 +11,163 @@
-                        if (index < word.length() && word[index] == '.')
-                        { // if real
-                            int oldIndex = index;
-                            index++;
-                            std::string floatPart = readWhileNum(index, word);
-                            if (floatPart.length() == 0)
-                            { //eg ..
-                                index = oldIndex;
-                                result.push_back({type, lineNum, num});
-                                break;
-                            }
-                            num.append(".");
-                            num.append(floatPart);
-                            type = TokenType::RealLiteral;
-                        }
-                        result.push_back({type, lineNum, num});
-                    }
-                    else if (std::isalpha(c))
-                    { // if var name
-                        std::string name = toStr(c);
-                        while (index < word.length() && (std::isalpha(word[index]) || std::isdigit(word[index]) || word[index] == '_'))
-                        {
-                            name.push_back(word[index++]);
-                        }
-                        val = trie.Find(name).value_or(TokenType::InvalidToken);
-                        if (val == TokenType::InvalidToken)
-                        {
-                            result.push_back({TokenType::Identifier, lineNum, name});
-                        }
-                        else
-                        {
-                            result.push_back({val, lineNum, sign});
-                        }
-                    }
-                    else
-                    { // non-valid construction or space
-                        // std::cout << "INVALID CHAR" << c << "\n";
-                        if (!std::isspace(c))
-                        {
-                            result.push_back({TokenType::InvalidToken, lineNum, toStr(c)});
-                        }
-                    }
-
-                    break;
-                }
-            }
+    if (std::isdigit(m_buf[m_pos])) {
+
+        auto truncLen = skipWhile(m_pos, Lexer::isdigit, 1);
+
+        size_t fracLen = 0;
+        if (m_pos + truncLen < m_buf.size() && m_buf[m_pos + truncLen] == '.') {
+            fracLen = skipWhile(m_pos + truncLen + 1, Lexer::isdigit);
         }
-        return 0;
-    }
 
-    /**
-     * Processing a line of riddle code. It updates the 'result' vector of tokens
-     * 
-     * @param line, string with line
-    */
-    void processLine(std::string line)
-    {
-        lineNum++;
-        std::istringstream iss(line);
-        std::string word;
-        while (std::getline(iss, word, ' '))
-        {
-            if (processWord(word) == -1)
-            { //if get a single-line comment
-                return;
-            }
+        if (fracLen != 0) {
```

Comment:
What if the input was `5.`? It should be interpreted as a `RealLiteral`, but it has no `fracLen`

Reply:
On a related note, I don't think the complementary case (`.5`) is handled as well

Reply:
You're right. When writing this I decided to abandon these real literals. But I guess, they're not that hard to incorporate.